doc/deployment/openshift.md

```diff
@@ -1,112 +1,384 @@
-              }
-            ],
+Below, find an example is using AWS elastic block storage as a persistent disk with a custom deploy.
+We'll show how to remove this PV in case you want to use a PV you create separately.
 
-     ... <snip>  ...
 
-          "spec": {
-            "volumes": [
-              {
-                "name": "etcd-storage",
-                "emptyDir": {}
-              }
-            ],
-    ```
+```
+pachctl deploy custom --persistent-disk aws --object-store s3 <pv-storage-name> <pv-storage-size> <s3-bucket-name> <s3-access-key-id> <s3-access-secret-key> <s3-access-endpoint-url> --static-etcd-volume=<pv-storage-name> > manifest.json
+```
 
-    Please note that `emptyDir` does not persist your data. You need to configure persistent volume or hostPath to persist your data.
+### 3. Modify pachd Service ports
 
+In the deployment manifest, which we called `manifest.json`, above, find the stanza for the `pachd` Service.  An example is shown below.
+```
+{
+	"kind": "Service",
+	"apiVersion": "v1",
+	"metadata": {
+		"name": "pachd",
+		"namespace": "default",
+		"creationTimestamp": null,
+		"labels": {
+			"app": "pachd",
+			"suite": "pachyderm"
+		},
+		"annotations": {
+			"prometheus.io/port": "9091",
+			"prometheus.io/scrape": "true"
+		}
+	},
+	"spec": {
+		"ports": [
+			{
+				"name": "s3gateway-port",
+				"port": 600,
+				"targetPort": 0,
+				"nodePort": 30600
+			},
+			{
+				"name": "api-grpc-port",
+				"port": 650,
+				"targetPort": 0,
+				"nodePort": 30650
+			},
+			{
+				"name": "trace-port",
+				"port": 651,
+				"targetPort": 0,
+				"nodePort": 30651
+			},
+			{
+				"name": "api-http-port",
+				"port": 652,
+				"targetPort": 0,
+				"nodePort": 30652
+			},
+			{
+				"name": "saml-port",
+				"port": 654,
+				"targetPort": 0,
+				"nodePort": 30654
+			},
+			{
+				"name": "api-git-port",
+				"port": 999,
+				"targetPort": 0,
+				"nodePort": 30999
+			}
+		],
+		"selector": {
+			"app": "pachd"
+		},
+		"type": "NodePort"
+	},
+	"status": {
+		"loadBalancer": {}
+	}
+}
+```
+While the nodePort declarations are fine, the port declarations are too low for OpenShift. Good example values are shown below.
+```
+	"spec": {
+		"ports": [
+			{
+				"name": "s3gateway-port",
+				"port": 1600,
+				"targetPort": 0,
+				"nodePort": 30600
+			},
+			{
+				"name": "api-grpc-port",
+				"port": 1650,
+				"targetPort": 0,
+				"nodePort": 30650
+			},
+			{
+				"name": "trace-port",
+				"port": 1651,
+				"targetPort": 0,
+				"nodePort": 30651
+			},
+			{
+				"name": "api-http-port",
+				"port": 1652,
+				"targetPort": 0,
+				"nodePort": 30652
+			},
+			{
+				"name": "saml-port",
+				"port": 1654,
+				"targetPort": 0,
+				"nodePort": 30654
+			},
+			{
+				"name": "api-git-port",
+				"port": 1999,
+				"targetPort": 0,
+				"nodePort": 30999
+			}
+		],
+```
 
-3. Deploy Pachyderm manifest you modified.
+### 4. Modify pachd Deployment ports and add environment variables
```

Comment:
Overall, I think it would be better if we encouraged people to use the [OCPify script](https://gist.github.com/gabrielgrant/86c1a5b590ae3f4b3fd32d7e9d622dc8) (perhaps along with a modified version of the wrapper script written for the original OpenShift demo customer) to make these edits to their manifest in an automated way. Making these edits manually, by hand, seems pretty error prone, difficult to keep track of, and will have to be re-done any time a user want to deploy to a new pach version.

That being said, it's certainly important to explain how all of this works, so people can understand the script/do it by hand if they want/have other requirements, but I don't think that should be the default we're suggesting.

Reply:
I think I did this sentiment justice!